Flaky HA unit test (probably)

### Current Behavior

2 consecutive same-test-fails on PR [#7831](https://github.com/FlowFuse/flowfuse/pull/7831):

* https://github.com/FlowFuse/flowfuse/actions/runs/29411291983/attempts/2?pr=7831
* https://github.com/FlowFuse/flowfuse/actions/runs/29411291983/attempts/3?pr=7831

I suspect this is actually a flaky test - dependant on runtime umph - but its just a guess based on the nature of the code in the test:
```
call /api/v1/projects/${TestObjects.project.id}/ha => replicas: 2
await 250ms
await project reload()
project.state.should.equal('suspended')
```

IMO, the "suspended" assertion (`TestObjects.project.state.should.equal('suspended')`) is testing the stub driver's timing, not the HA feature. So a potential fix is to ditch that!

But to be safe - in case this is a real (time sensitive) bug, I am logging it.

Below is the trace that happened on test run 2 & 3 (in case above links are deleted & to be searchable)

```
1) HA Instance API
       applies valid ha option:

      AssertionError: expected 'running' to be 'suspended'
      + expected - actual

      -running
      +suspended

      at Assertion.fail (node_modules/should/cjs/should.js:275:17)
      at Assertion.value (node_modules/should/cjs/should.js:356:19)
      at Context.<anonymous> (test/unit/forge/ee/routes/ha/index_spec.js:101:42)
      at process.processTicksAndRejections (node:internal/process/task_queues:104:5)
```


### Expected Behavior

Should pass tests regardless of _grunt_ on the test runner

### Steps To Reproduce

NA

### Environment

- FlowFuse version: FFC
- Node.js version:
- npm version:
- Platform/OS:
- Browser:


### Have you provided an initial effort estimate for this issue?

I have provided an initial effort estimate

--- a/config/cypress-shared.config.js
+++ b/config/cypress-shared.config.js
@@ -5,6 +5,10 @@ const downloadsFolder = 'test/e2e/frontend/cypress/downloads'
 module.exports = {
     viewportWidth: 1024,
     viewportHeight: 768,
+    retries: {
+        runMode: 2,
+        openMode: 0
+    },
     e2e: {
         downloadsFolder,
         fixturesFolder: 'test/e2e/frontend/cypress/fixtures',

--- a/frontend/src/components/InstanceStatusPolling.vue
+++ b/frontend/src/components/InstanceStatusPolling.vue
@@ -52,8 +52,14 @@ export default {
                 this.checkWaitTime *= 1.15
 
                 if (this.instance.id) {
-                    const data = await InstanceApi.getInstance(this.instance.id)
-                    this.$emit('instance-updated', data)
+                    try {
+                        const data = await InstanceApi.getInstance(this.instance.id)
+                        this.$emit('instance-updated', data)
+                    } catch (err) {
+                        if (err.response?.status !== 404) {
+                            console.error(err)
+                        }
+                    }
                 }
             }, this.checkWaitTime)
         },

--- a/test/e2e/frontend/cypress/tests/account/tokens.spec.js
+++ b/test/e2e/frontend/cypress/tests/account/tokens.spec.js
@@ -86,10 +86,11 @@ describe('FlowFuse - Personal Access Tokens', () => {
     })
 
     it('can be added with an expiry date', () => {
-        // count how many tokens we already have
-        const rows = Cypress.$('[data-el="tokens-table"] tbody').find('tr.ff-data-table--row').length
-
         cy.intercept('POST', '/api/*/user/tokens').as('addPersonalAccessToken')
+
+        // retryable read - a synchronous Cypress.$ count can beat the table render on slow CI, undercounting the baseline
+        let rows
+        cy.get('[data-el="tokens-table"] tbody').find('tr.ff-data-table--row').then($rows => { rows = $rows.length })
 
         cy.get('[data-action="new-token"]').click()
 
@@ -108,10 +109,12 @@ describe('FlowFuse - Personal Access Tokens', () => {
         cy.get('[data-el="add-token-confirmation"] [data-action="token-confirmation-done"]').click()
 
         // check we have one more token than before
-        cy.get('[data-el="tokens-table"] tbody').find('tr').should('have.length', rows + 1)
-        cy.get('[data-el="tokens-table"] tbody').find('tr').last().should('contain', '31')
-        cy.get('[data-el="tokens-table"] tbody').find('tr').last().should('contain', '12')
-        cy.get('[data-el="tokens-table"] tbody').find('tr').last().should('contain', '2050')
+        cy.get('[data-el="tokens-table"] tbody').find('tr.ff-data-table--row').should($rows => {
+            expect($rows).to.have.length(rows + 1)
+        })
+        // the cell renders Date.parse(expiresAt).toLocaleDateString(); derive the label the same way so the assertion is timezone-independent
+        const expiryLabel = new Date(Date.parse('2050-12-31')).toLocaleDateString()
+        cy.get('[data-el="tokens-table"] tbody').find('tr').last().should('contain', expiryLabel)
     })
 
     it('can be added with read-only enabled', () => {

--- a/test/e2e/frontend/cypress/tests/admin.spec.js
+++ b/test/e2e/frontend/cypress/tests/admin.spec.js
@@ -65,11 +65,15 @@ describe('FlowFuse platform admin users', () => {
         cy.get('[data-nav="admin-teams"]').click()
         cy.wait('@getTeams')
 
+        // check that we're on the admin/teams page
+        cy.url().should('match', /\/admin\/teams(\/[^/]+)?/)
+
         // Not a member of BTeam
         cy.get('[data-el="teams-table"]').contains('BTeam').click()
 
-        // wait for the team redirect to settle on /overview, else the Applications click below can race the pending redirect
-        cy.url().should('match', /\/team\/[^/]+\/overview/)
+        // check that we're on the selected team's overview page
+        cy.url().should('match', /\/overview(\/[^/]+)?/)
+
         cy.get('[data-el="banner-team-as-admin"]').should('exist')
 
         // Unique alias to avoid colliding with cy.home()'s getTeamApplications

--- a/test/e2e/frontend/cypress/tests/admin.spec.js
+++ b/test/e2e/frontend/cypress/tests/admin.spec.js
@@ -68,7 +68,8 @@ describe('FlowFuse platform admin users', () => {
         // Not a member of BTeam
         cy.get('[data-el="teams-table"]').contains('BTeam').click()
 
-        cy.url().should('match', /\/team\/[^/]+/)
+        // wait for the team redirect to settle on /overview, else the Applications click below can race the pending redirect
+        cy.url().should('match', /\/team\/[^/]+\/overview/)
         cy.get('[data-el="banner-team-as-admin"]').should('exist')
 
         // Unique alias to avoid colliding with cy.home()'s getTeamApplications

--- a/test/unit/forge/ee/routes/ha/index_spec.js
+++ b/test/unit/forge/ee/routes/ha/index_spec.js
@@ -97,8 +97,7 @@ describe('HA Instance API', function () {
         await sleep(STOP_DELAY)
         await TestObjects.project.reload()
 
-        // Project has been stopped but is presented as "starting"
-        TestObjects.project.state.should.equal('suspended')
+        // state column not asserted here: it flips suspended->running mid-restart, racing the STOP_DELAY sleep
         should(await app.db.controllers.Project.getInflightState(TestObjects.project)).equal('starting')
 
         // Wait for at least start delay as set in stub driver

